fuel-metrics/src/p2p_metrics.rs

```diff
@@ -0,0 +1,48 @@
+use lazy_static::lazy_static;
+use once_cell::race::OnceBox;
+use prometheus_client::{
+    metrics::counter::Counter,
+    registry::Registry,
+};
+use std::{
+    boxed::Box,
+    default::Default,
+};
+
+pub struct P2PMetrics {
+    pub gossip_sub_registry: OnceBox<Registry>,
+    pub peer_metrics: Registry,
+    pub unique_peers: Counter,
+}
+
+impl Default for P2PMetrics {
+    fn default() -> Self {
+        let peer_metrics = Registry::default();
+
+        let unique_peers = Counter::default();
+
+        Self {
+            gossip_sub_registry: OnceBox::new(),
+            peer_metrics,
+            unique_peers,
+        }
+    }
+}
+
+pub fn init(mut metrics: P2PMetrics) -> P2PMetrics {
```

Comment:
Seems it can be easy part of `P2PMetrics` as a consturctor=)

Reply:
It seems the private `new` method feets better than the `Default` because we expect only one instance of the metrics.

Reply:
Good idea, will change initialization from `default` to `new`